src/DataWrangling/ECCO/ECCODarwin.jl

```diff
@@ -0,0 +1,485 @@
+Base.iterate(metadata::ECCODarwinMetadata{<:AbstractCFDateTime}) = (metadata, nothing)
+Base.iterate(::ECCODarwinMetadata{<:AbstractCFDateTime}, ::Any)  = nothing
+
+Base.length(metadata::ECCODarwinMetadata) = length(metadata.dates)
+Base.size(data::ECCODarwinMetadata{<:Any, <:Any, <:ECCO270DarwinMonthly}) = (1080, 540, 50, length(data.dates))
+Base.size(data::ECCODarwinMetadata{<:Any, <:Any, <:ECCO4DarwinMonthly})   = (360,  180, 50, length(data.dates))
+
+Base.length(metadata::ECCODarwinMetadata{<:AbstractCFDateTime}) = 1
+Base.size(::ECCODarwinMetadata{<:AbstractCFDateTime, <:Any, <:ECCO270DarwinMonthly}) = (1080, 540, 50, 1)
+Base.size(::ECCODarwinMetadata{<:AbstractCFDateTime, <:Any, <:ECCO4DarwinMonthly})   = (360,  180, 50, 1)
+
+# The whole range of dates in the different dataset versions
+all_ECCO_dates(::ECCO4DarwinMonthly)   = DateTimeProlepticGregorian(1992, 1, 1) : Month(1) : DateTimeProlepticGregorian(2023, 12, 31)
+all_ECCO_dates(::ECCO270DarwinMonthly) = DateTimeProlepticGregorian(1992, 1, 1) : Month(1) : DateTimeProlepticGregorian(2023, 12, 31)
+
+# File names of metadata containing multiple dates
+metadata_filename(metadata) = [metadata_filename(metadatum) for metadatum in metadata]
+
+# File name generation specific to each Dataset version
+function metadata_filename(metadata::ECCODarwinMetadata{<:AbstractCFDateTime})
+    shortname = short_name(metadata)
+    
+    iternum = Dates.value((metadata.dates-metadata.date_ref)/(metadata.dt*1e3))
+    iterstr = string(iternum, pad=10)
+
+    return shortname * "." * iterstr * ".data"
+end
+
+# Convenience functions
+metadata_path(metadata) = joinpath(metadata.dir, metadata_filename(metadata))
+short_name(data::ECCODarwinMetadata{<:Any, <:Any, <:ECCO270DarwinMonthly}) = ECCO270Darwin_short_names[data.name]
+short_name(data::ECCODarwinMetadata{<:Any, <:Any, <:ECCO4DarwinMonthly})   = ECCO4Darwin_short_names[data.name]
+
+metadata_url(prefix, m::ECCODarwinMetadata{<:Any, <:Any, <:ECCO4DarwinMonthly}) = prefix * "/" * short_name(m) * "/" * metadata_filename(m)
+metadata_url(prefix, m::ECCODarwinMetadata{<:Any, <:Any, <:ECCO270DarwinMonthly}) = prefix * "/" * short_name(m) * "/" * metadata_filename(m)
+
+location(data::ECCODarwinMetadata) = ECCODarwin_location[data.name]
+
+variable_is_three_dimensional(data::ECCODarwinMetadata) =
+    data.name == :DIC ||
+    data.name == :ALK ||
+    data.name == :PO₄ ||
+    data.name == :NO₃ ||
+    data.name == :DOP ||
+    data.name == :POP ||
+    data.name == :Fe  ||
+    data.name == :Siᵀ
+
+ECCO4Darwin_short_names = Dict(
+    :DIC => "DIC",
+    :ALK => "ALK",
+    :PO₄ => "PO4",
+    :NO₃ => "NO3",
+    :DOP => "DOP",
+    :POP => "POP",
+    :Fe  => "FeT",
+    :Siᵀ => "SiO2",
+    :PAR => "PAR",
+)
+
+ECCO270Darwin_short_names = Dict(
+    :DIC => "DIC",
+    :ALK => "ALK",
+    :PO₄ => "PO4",
+    :NO₃ => "NO3",
+    :DOP => "DOP",
+    :POP => "POP",
+    :Fe  => "FeT",
+    :Siᵀ => "SiO2",
+    :PAR => "PAR",
+)
```

Comment:
This seems to be the same as `ECCO4Darwin_short_names` so it might make sense to define one single `ECCODarwin_short_names`